src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantOverlapAnnotator.java

```diff
@@ -156,26 +158,39 @@ public VariantContext annotateOverlap(final List<VariantContext> overlapTestVCs,
     private static String getRsID(final List<VariantContext> rsIDSourceVCs, final VariantContext vcToAnnotate) {
         Utils.nonNull(rsIDSourceVCs, "rsIDSourceVCs cannot be null");
         Utils.nonNull(vcToAnnotate, "vcToAnnotate cannot be null");
+        final List<String> rsids = new ArrayList<>();
 
-        for ( final VariantContext vcComp : rsIDSourceVCs ) {
-            if ( vcComp.isFiltered() ) {
+        final List<VariantContext> vcAnnotateList = GATKVariantContextUtils.splitVariantContextToBiallelics(vcToAnnotate, true,
+                GenotypeAssignmentMethod.SET_TO_NO_CALL_NO_ANNOTATIONS, true);
+
+        for ( final VariantContext vcCompSource : rsIDSourceVCs ) {
+            if ( vcCompSource.isFiltered() ) {
                 continue; // don't process any failed VCs
             }
 
-            if ( ! vcComp.getContig().equals(vcToAnnotate.getContig()) || vcComp.getStart() != vcToAnnotate.getStart() ) {
-                throw new IllegalArgumentException("source rsID VariantContext " + vcComp + " doesn't start at the same position as vcToAnnotate " + vcToAnnotate);
+            if (!vcCompSource.getContig().equals(vcToAnnotate.getContig())) {
+                throw new IllegalArgumentException("source rsID VariantContext " + vcCompSource + " is not on same chromosome as vcToAnnotate " + vcToAnnotate);
```

Comment:
Hmm... splitVariantContextToBiallelics() claims that under some circumstances it results in variants getting pushed forwards on the genome. This is all fine and well but it leaves a hole here where there could be a variant in the input (or the DBSNP) that gets moved and would otherwise match with the next variant in the other file. Not that the old code could handle this case any better... This is a very niche circumstance and probably doesn't warrant worrying about. I do think we should probably document this fact in a comment somewhere.

Reply:
I agree that there are still cases that this solution will not catch, but I think they are related to left alignment of indels or MNP vs multiple SNP/ complex snp/indel combination issues, not the allele movement that occurs in splitVariantContextToBiallelics.  Since there is no reference passed to splitVariantContextToBiallelics() beyond what is stored in the reference allele of the variant context, the alleles that are returned by this method cannot overlap any reference bases that were not overlapped by the original alleles.  So I think any match that would be found if the splitting and trimming were performed before grabbing the overlaps will also be found by this code.

But I agree with the general point, so I have added a comment that in rare cases we may fail add annotations to a variant that "should" have been added.